libs/shared/console-shared/src/lib/git-repository-settings/feature/edit-git-repository-settings-feature/edit-git-repository-settings-feature.tsx

```diff
@@ -1,31 +1,27 @@
-import { useCallback, useEffect, useState } from 'react'
+import { useEffect, useState } from 'react'
 import { useFormContext } from 'react-hook-form'
-import { useSelector } from 'react-redux'
 import { useParams } from 'react-router-dom'
-import { selectApplicationById } from '@qovery/domains/application'
-import { isJob, isJobGitSource } from '@qovery/shared/enums'
-import { type ApplicationEntity } from '@qovery/shared/interfaces'
+import { match } from 'ts-pattern'
+import { useService } from '@qovery/domains/services/feature'
+import { isHelmGitSource, isJobGitSource } from '@qovery/shared/enums'
 import { upperCaseFirstLetter } from '@qovery/shared/util-js'
-import { type RootState } from '@qovery/state/store'
 import GitRepositorySettings from '../../ui/git-repository-settings/git-repository-settings'
 
-export function EditGitRepositorySettingsFeature() {
-  const { applicationId = '' } = useParams()
+export interface EditGitRepositorySettingsFeatureProps {
+  withBlockWrapper?: boolean
+}
 
-  const application = useSelector<RootState, ApplicationEntity | undefined>(
-    (state) => selectApplicationById(state, applicationId),
-    (a, b) =>
-      JSON.stringify(a?.git_repository) === JSON.stringify(b?.git_repository) ||
-      (isJobGitSource(a?.source) &&
-        isJobGitSource(b?.source) &&
-        JSON.stringify(a?.source?.docker?.git_repository) === JSON.stringify(b?.source?.docker?.git_repository))
-  )
+export function EditGitRepositorySettingsFeature({ withBlockWrapper = true }: EditGitRepositorySettingsFeatureProps) {
+  const { applicationId = '' } = useParams()
+  const { data: service } = useService({ serviceId: applicationId })
 
-  const getGitRepositoryFromApplication = useCallback(() => {
-    return isJob(application) && isJobGitSource(application?.source)
-      ? application?.source?.docker?.git_repository
-      : application?.git_repository
-  }, [application])
+  const gitRepository = match(service)
+    .with({ serviceType: 'JOB' }, (job) => (isJobGitSource(job.source) ? job.source.docker?.git_repository : undefined))
+    .with({ serviceType: 'APPLICATION' }, (application) => application.git_repository)
+    .with({ serviceType: 'HELM' }, (helm) =>
```

Comment:
```suggestion
    .with({ serviceType: 'HELM', source: P.when(isHelmGitSource) }, (helm) =>
```
Then you can remove the ternary inside.